### PR TITLE
Create and delete companie

### PR DESCRIPTION
**Pull request by ducnghia1205**

### CODE REVIEW

File: app/companies/controller.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+      return res.json(result.rows);
+    } catch (e) {
+      console.log(e.message)
+    }
+  },
+  createCompany: async (req, res) => {
+    try {
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
```
Comment (phatnhse) [inline app/companies/controller.js:28]: Duplicated method here, we can create a method for this check to reduce the boilerplate code

File: app/companies/controller.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  getListCompanies: async (req, res) => {
+    try {
+      const perPage = req.query.per_page || 20;
+      const page = req.query.page || 1;
+      const result = await PSQL.query(
+        `SELECT * FROM companies LIMIT ${perPage} OFFSET ${(page - 1) * perPage}`
+      );
+
+      if (!result) {
+        return res.send('No have data!')
```
Comment (ducnghia1205) [inline app/companies/controller.js]: The message content shoud be: `No data available`

File: app/companies/controller.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      return res.json('success');
+    } catch (e) {
+      console.log(e.message);
+    }
+  },
+  updateCompany: async (req, res) => {
+    try {
+      const id = req.params.id;
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
```
Comment (ducnghia1205) [inline app/companies/controller.js:44]: same here

File: app/companies/controller.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        return res.status(422).json({errors: errors.array()});
+      }
+      const company = await db('companies').select('*').where('id', id);
+      if (!company || !company.length) {
+        return res.json(`company does't exist`);
+      }
+      req.body.updated_at = moment().format('YYYY-MM-DD HH:mm:ss.ssssssZZ').slice(0, -2);
```
Comment (ducnghia1205) [inline app/companies/controller.js:51]: can we export this function as a `DateTimeUtils` method

File: app/companies/controller.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      return res.json('success');
+    } catch (e) {
+      console.log(e.message);
+    }
+  },
+  deleteCompany: async (req, res) => {
+    try {
+      const id = req.params.id;
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
```
Comment (ducnghia1205) [inline app/companies/controller.js:62]: same here

File: app/companies/controller.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        return res.status(422).json({errors: errors.array()});
+      }
+      const company = await db('companies').select('*').where('id', id);
+      if (!company || !company.length) {
+        return res.json(`company does't exist`);
+      }
+      req.body.updated_at = moment().format('YYYY-MM-DD HH:mm:ss.ssssssZZ').slice(0, -2);
+      await db('companies').update(req.body).where('id', id);
+
+      return res.json('success');
```
Comment (ducnghia1205) [inline app/companies/controller.js]: format json to return the updated record

File: app/companies/controller.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  },
+  createCompany: async (req, res) => {
+    try {
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
+        return res.status(422).json({errors: errors.array()});
+      }
+      await db('companies').insert(req.body);
+
+      return res.json('success');
```
Comment (ducnghia1205) [inline app/companies/controller.js]: format json to return the recent created company

File: migrations/20190817144117_createCompanies.js [hunk 8]
```diff
@@ -28,6 +28,10 @@ exports.up = async function (knex) {
     table.string('founded_at')
       .nullable()
 
+    table.timestamp('deleted_at')
+      .nullable()
+      .defaultTo()
```
Comment (ducnghia1205) [inline migrations/20190817144117_createCompanies.js]: do we need defaultTo :thinking: ?

File: app/companies/controller.js (hunk 1 above)
Comment (ducnghia1205) [reply app/companies/controller.js:28]: create new function will be have the same lines here

**Pull request closed without merging**